### ISSUE TITLE
dists.json and patch for B::C and 5.18.4

### ISSUE THREAD

**Issue by hartzell** [iovation/rpmcpan#4]
I had trouble getting B::C to pass t/perldoc.t, I was seeing the same problem that was reported on [cpantesters](http://www.cpantesters.org/cpan/report/8701c360-3858-11e4-93a3-ef93dfbfc7aa).

It seems to be due to confusion between the version of Cwd in the 5.18.4 base system and the newer version that some module pulled in during the  build.

```
#   Failed test 'same result'
#   at t/perldoc.t line 102.
#          got: 'Cwd object version 3.40 does not match $Cwd::VERSION 3.47 at (eval 11) line 68.
# '
#     expected: 'No documentation found for "-T".
```

I did not see the failure in `t/issue24.t` that cpantesters saw.

I fixed it with a big hammer:

```
--- a/t/perldoc.t   2015-05-22 21:34:22.943582144 +0000
+++ b/t/perldoc.t   2015-05-22 21:34:33.579898843 +0000
@@ -1,6 +1,6 @@
 #! /usr/bin/env perl
 # brian d foy: "Compiled perlpod should be faster then uncompiled"
-use Test::More;
+use Test::More skip_all => 'Skip failing tests, conflict w/ dual-lived Cwd modules';
 use strict;
 BEGIN {
   unshift @INC, 't';
```

Not sure if there's a similar problem with a more current perl release and/or if this patch is worth conditionalizing somehow or.....

**Comment by theory** [comment]
I dunno, but you can supply patches by putting them into the `SOURCES` directory and setting them for a particular distribution in your `dists.json` file. I've done that for CGI-Compile [here](https://github.com/iovation/rpmcpan/blob/master/etc/dists.json#L29), for example. Note that patches are applied via `patch -p1`; your patch above looks right for that.

**Comment by hartzell** [comment]
Sorry, I wasn't clear.  I did just what you suggest above, have that patch in my SOURCES and a line like you describe in my etc/dists.json.  Here's a retelling with sufficient AM coffee.    I haven't tried doing a build w/ a newer perl to see if the patch is useful/painful/neutral there so I'm not particularly suggesting that it's something you want to include.  I guess I'm mostly interested in leaving an Internet breadcrumb in case someone else trips over the same problem (paying forward).

When building rpm's using 5.18.4, B::C has trouble passing it's tests, due to what appears to be a conflict between the version of Cwd that is included in 5.18.4's core and the more recent version that the build process installs.  Here's a link to a similar failure on [cpantesters](http://www.cpantesters.org/cpan/report/8701c360-3858-11e4-93a3-ef93dfbfc7aa) (I did not see the failure in t/issue24.t that cpantesters saw).

I was able to an rpm for the distribution that provides B::C by including this line in my etc/dists.json

```
    "B-C": { "patch": ["b-c-perldoc.t.patch"] },
```

and this patch in SOURCES/b-c-perldoc.t.patch:

```
--- a/t/perldoc.t   2015-05-22 21:34:22.943582144 +0000
+++ b/t/perldoc.t   2015-05-22 21:34:33.579898843 +0000
@@ -1,6 +1,6 @@
 #! /usr/bin/env perl
 # brian d foy: "Compiled perlpod should be faster then uncompiled"
-use Test::More;
+use Test::More skip_all => 'Skip failing tests, conflict w/ dual-lived Cwd modules';
 use strict;
 BEGIN {
   unshift @INC, 't';
```

Thanks again for sharing the tool, it's been phenomenally helpful.

**Comment by theory** [comment]
Ah, looks good. I have quite a few similar patches in `SOURCES`.
